pkg/registrar/registrar.go

```diff
@@ -98,7 +98,12 @@ func (r *Registrar) GetNames(key string) ([]string, error) {
 	if !exists {
 		return nil, ErrNoSuchKey
 	}
-	return names, nil
+
+	ls := make([]string, 0, len(names))
+	for _, n := range names {
```

Comment:
No real reason other than I've had the results of `copy` suprise me more than once.